### PR TITLE
Added connection to mysql via socket

### PR DESCRIPTION
**Pull request by IgorKalinchuk**

### CODE REVIEW

File: lib/api.db.js [hunk 1]
```diff
@@ -75,7 +75,7 @@ api.db.openApplicationDatabases = function(application, callback) {
 api.db.openDatabase = function(application, database, callback) {
   if (!application.databases) application.databases = {};
   database.application = application;
-  database.schema = database.url.substr(0, database.url.indexOf(':'));
+  database.schema = (database.url) ? database.url.substr(0, database.url.indexOf(':')) : 'mysql';
```
Comment (tshemsedinov) [inline lib/api.db.js]: mysql by default, why not mongo or pg?

File: lib/api.db.mysql.js [hunk 2]
```diff
@@ -28,7 +28,7 @@ if (api.mysql) {
   api.db.mysql.open = function(database, callback) {
     database.retryCounter++;
     var application = database.application,
-        connection = api.mysql.createConnection(database.url);
```
Comment (tshemsedinov) [inline lib/api.db.mysql.js]: Everything should be connection in connection string url

Comment (IgorKalinchuk) [inline lib/api.db.mysql.js]: When msql configured with closed TCP/IP. I needed to use connection via socet, so I can't use string url in this case.
socketPath: The path to a unix domain socket to connect to. When used host and port are ignored.

File: lib/api.db.js (hunk 1 above)
Comment (IgorKalinchuk) [inline lib/api.db.js]: Yes, but as I understand only in mysql, can't be used string url

**Pull request closed without merging**